internal/patroni/config.go

```diff
@@ -246,9 +246,9 @@ func DynamicConfiguration(
 	postgresql["parameters"] = parameters
 
 	// Copy the "postgresql.pg_hba" section after any mandatory values.
-	hba := make([]string, len(pgHBAs.Mandatory))
+	hba := make([]string, 0)
```

Comment:
❓ We should allocate when possible. Does the following quiet the linter as well?
```suggestion
	hba := make([]string, 0, len(pgHBAs.Mandatory))
```